nearpy/storage/storage_redis.py

```diff
@@ -86,6 +86,49 @@ def store_vector(self, hash_name, bucket_key, v, data):
         # Push JSON representation of dict to end of bucket list
         self.redis_object.rpush(redis_key, pickle.dumps(val_dict, protocol=2))
 
+    def store_many_vectors(self, hash_name, bucket_keys, vs, data):
+        """
+        Store a batch of vectors in Redis.
+        Stores vector and JSON-serializable data in bucket with specified key.
+        """
+        with self.redis_object.pipeline() as pipeline:
+            for idx, v in enumerate(vs):
+                redis_key = self._format_redis_key(hash_name, bucket_keys[idx])
```

Comment:
I suggest extracting internal method `_add_vector(self, hash_name, bucket_keys, vs, data, redis)` and using it as
```
def store_vector(self, hash_name, bucket_key, v, data):
    self._add_vector(hash_name, bucket_key, v, data, self.redis)

def store_many_vectors(self, hash_name, bucket_keys, vs, data):
    with self.redis_object.pipeline() as pipeline:
        for bucket_key, v in zip(bucket_keys, vs):
            self._add_vector(hash_name, bucket_key, v, data, pipeline)
        pipeline.execute()
```
It should help with code duplication.

Reply:
Sorry for the long delay. I haven't added unit tests for this PR. Pls give me one week's time, i will finish it up.

Reply:
@pixelogik Two of the issues I pointed out in comments are still here and I would like to see them fixed.

Reply:
@amorgun Thanks for the feedback. It's in progress.

Reply:
Thanks for this suggestion. I included this modification in commit `4e001a1`